### ISSUE TITLE
YAML script for VPC service controls 

### ISSUE THREAD

**Issue by vivek725k** [GoogleCloudPlatform/k8s-config-connector#658]
## Bug Description##

Getting error while crating YAML script for VPC service controls , 

**below are the code-**

apiVersion: accesscontextmanager.cnrm.cloud.google.com/v1beta1

kind: AccessContextManagerServicePerimeter

metadata:

  name: tmoaccessperimeter

  namespace: ccoe # kpt-set: ${ccoe-namespace}

  annotations:

    cnrm.cloud.google.com/project-id: "project-id" # kpt-set: ${project-id}

spec:

  spec:  

    resources:

    - projectRef:

        external: projects/464958446245 # kpt-set: projects/${project-number}

    restrictedServices:

    - storage.googleapis.com

    - cloudkms.googleapis.com

    - compute.googleapis.com

    - secretmanager.googleapis.com

    ingressPolicies:

    - ingressFrom:

        identities:

        - serviceAccountRef:

            external: ccoe-sa@prj-hosted-prd-1.iam.gserviceaccount.com

        sources:

        - projectRef:

            external: projects/621288388383

      ingressTo:

        operations:

        - methodSelectors:

          - method: "*"

          serviceName: storage.googleapis.com

        resources:

        - projectRef:

            external: projects/464958446245

  accessPolicyRef:

    external: accessPolicies/964444541971

  useExplicitDryRunSpec: true

  title: Service Perimeter

  perimeterType: PERIMETER_TYPE_REGULAR

**Error-** 

Status:Conditions:Last Transition Time: 2022-05-02T13:59:20ZMessage: Update call failed: error fetching live state: error converting resource config: expected reference identities to be object but was notReason: UpdateFailedStatus: FalseType: ReadyObserved Generation: 6Events:Type Reason Age From Message---- ------ ---- ---- -------Warning UpdateFailed 16s (x6 over 74s) accesscontextmanagerserviceperimeter-controller Update call failed: error fetching live state: error converting resource config: expected reference identities to be

**Comment by vivek725k** [comment]
please update on this, waiting for your response.

**Comment by caieo** [comment]
Hi @vivek725k, can we ask that you fill out the bug template completely? We rely on some critical details asked by that template rather than typing it out every time we receive a new bug request. The error message you posted also is not complete. 

Also in the future, can you please properly format your YAML snippets? It makes it very difficult to debug issues when the YAML snippets are not formatted properly. Thank you.

**Comment by Vanditha-V-R** [comment]
<img width="397" alt="VPCSC" src="https://user-images.githubusercontent.com/104434780/166667592-24f1ab6b-68f4-4c7a-8ae0-7be08521ff1b.PNG">

**Comment by Vanditha-V-R** [comment]
Hi,

We corrected that and getting still error.

Vanditha
________________________________

**Comment by maqiuyujoyce** [comment]
Thanks for the updates, @Vanditha-V-R .

Could you confirm if your error is still like below, or have you seen other errors?

```
Status:Conditions:Last Transition Time: 2022-05-02T13:59:20ZMessage: Update call failed: error fetching live state: error converting resource config: expected reference identities to be object but was notReason: UpdateFailedStatus: FalseType: ReadyObserved Generation: 6Events:Type Reason Age From Message---- ------ ---- ---- -------Warning UpdateFailed 16s (x6 over 74s) accesscontextmanagerserviceperimeter-controller Update call failed: error fetching live state: error converting resource config: expected reference identities to be
```

**Comment by mikesharpton-bestbuy** [comment]
Hello, I can confirm we are seeing the same errors with our config connector instance.  I was glad to see this issue as I had tried many times to get this to work.

`Warning  UpdateFailed  6s (x4 over 20s)  accesscontextmanagerserviceperimeter-controller  Update call failed: error fetching live state: error converting resource config: expected reference identities to be object but was not`

**Comment by mikesharpton-bestbuy** [comment]
If I remark ingressPolicies from my yaml, I can create a perimeter via config connector.  With it there, it fails.

`---
apiVersion: accesscontextmanager.cnrm.cloud.google.com/v1beta1
kind: AccessContextManagerServicePerimeter
metadata:
  name: serviceperimeter01
spec:
  spec:
    resources:
      - projectRef:
          external: "projects/projectid"
    restrictedServices:
      - "storage.googleapis.com"
    # ingressPolicies:
    # - ingressFrom:
    #     identities:
    #     - serviceAccountRef:
    #         external: "serviceAccount:sa@project"
    #     sources:
    #     - accessLevelRef:
    #         name: accesslevel01
    #   ingressTo:
    #     resources:
    #     - projectRef:
    #         external: "projects/projectid"
  status:
    resources:
      - projectRef:
          external: "projects/projectid"
    restrictedServices:
      - "storage.googleapis.com"
    # ingressPolicies:
    # - ingressFrom:
    #     identities:
    #     - serviceAccountRef:
    #         external: "serviceAccount:serviceAccount:sa@project"
    #     sources:
    #       - accessLevelRef:
    #           name: ccesslevel01
    #   ingressTo:
    #     resources:
    #     - projectRef:
    #         external: "projects/projectid"
  useExplicitDryRunSpec: true
  title: Service Perimeter
  accessPolicyRef:
    external: accessPolicies/652749556735
  description: A test service permimeter
  perimeterType: PERIMETER_TYPE_REGULAR`

**Comment by maqiuyujoyce** [comment]
Hi @mikesharpton-bestbuy thank you for sharing the information.

I was able to reproduce it and have found the root cause. Will update the issue when the fix is out.

**Comment by maqiuyujoyce** [comment]
Hi @tedelwartowski-bestbuy , the fix should be included in the next release, and we expect it to be out by the end of this week.

**Comment by maqiuyujoyce** [comment]
Hi, the fix has been released in [v1.85.0](https://github.com/GoogleCloudPlatform/k8s-config-connector/releases/tag/v1.85.0). Please let us know if you still have any issues using `AccessContextManagerServicePerimeter`.

**Comment by mikesharpton-bestbuy** [comment]
Thank you we will upgrade and test.  If you hear nothing, it is working.  Have a good one.
